src/channel.ts

```diff
@@ -1336,7 +1342,12 @@ export class Channel<StreamChatGenerics extends ExtendableGenerics = DefaultGene
     }
 
     // any event can send over the online count
-    if (event.watcher_count !== undefined) {
+    if (
+      event.type !== 'connection.changed' &&
+      event.type !== 'connection.recovered' &&
+      event.type !== 'transport.changed' &&
```

Comment:
We can extract this into a function `isLocalEvent`

Reply:
Great suggestion! Done 👍